src/pal/src/file/directory.cpp

```diff
@@ -609,13 +600,6 @@ CreateDirectoryA(
     // Canonicalize the path so we can determine its length.
     FILECanonicalizePath(realPath);
```

Comment:
Path canonicalization removes artefacts like `/./` `/../` or `//` from the path and I am not sure if mkdir handles those correctly.